src/Configuration/Config.AzureKeyVault/src/AzureKeyVaultConfigurationProvider.cs

```diff
@@ -36,14 +36,18 @@ public AzureKeyVaultConfigurationProvider(IKeyVaultClient client, string vault,
             _client = client ?? throw new ArgumentNullException(nameof(client));
             _vault = vault ?? throw new ArgumentNullException(nameof(vault));
             _manager = manager ?? throw new ArgumentNullException(nameof(manager));
-            if (reloadInterval != null && reloadInterval.Value <= TimeSpan.Zero)
+
+            if (reloadInterval != null)
             {
-                throw new ArgumentOutOfRangeException (nameof(reloadInterval), reloadInterval, nameof(reloadInterval) + " must be positive.");
+                if (reloadInterval.Value <= TimeSpan.Zero)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(reloadInterval), reloadInterval, nameof(reloadInterval) + " must be positive.");
+                }
+                _reloadInterval = reloadInterval.Value;
             }
 
             _pollingTask = null;
```

Comment:
BTW: this can be omitted, as in .NET fields are initialized with `default`, which is for referency types `null`.
```suggestion

```

Reply:
Corrected.